### DISCUSSION TITLE
Release Notes after Dev to Master Merge

### DISCUSSION THREAD

**Discussion by metda11** [rfennell/AzurePipelines#2014] [Q&A]
Hi everyone,

Is there a way to generate release notes that include only the newly assigned work items after merging from dev to master using the Generate Release Notes (Crossplatform) tool?

Our workflow:

When we need to fix something in our customers' live environment, we create a bug.
We start by creating a branch from master and, after finishing the fix, we create a PR into master, release, and dev.
This process works fine for us.

The issue:

When we make a release after a dev to master merge, we receive all the work items that were included in previous releases, due to the merged PRs from the dev branch. This clutter makes it hard to identify the actual new changes.

My question:

Is there a way to generate release notes that show only the new work items since the last successfull release, excluding those from previous releases that are part of the merged dev branch?

Appreciate any help or ideas on this!

**Comment by rfennell** [comment]
The limitation is that WI are retrieved using a call to the Azure DevOps REST API that returns the WI associated between two builds of the same definition

A technique to select the build range that has worked in the past is to have a `stage` in your multistage YAML that is only triggered when there is a run only when you do the specific type of release you require release notes form. In this `stage` add a `job` that runs the release notes making sure you set the `checkstage:true`. This should list all the WI since the last successful release to this stage.

The alternate is to generate your WI with a WIQL query

The [WIKI](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task#usage-patterns) has information of both techniques
